src/services/upload/uploader.ts

```diff
@@ -0,0 +1,83 @@
+import { Collection } from 'services/collectionService';
+import { File } from 'services/fileService';
+import { sleep } from 'utils/common';
+import { handleError, CustomError } from 'utils/common/errorUtil';
+import { decryptFile } from 'utils/file';
+import { logError } from 'utils/sentry';
+import { fileAlreadyInCollection } from 'utils/upload';
+import NetworkClient from './networkClient';
+import UIService from './uiService';
+import { FileUploadResults } from './uploadManager';
+import UploadService, {
+    BackupedFile,
+    EncryptedFile,
+    FileInMemory,
+    UploadFile,
+} from './uploadService';
+
+const TwoSecondInMillSeconds = 2000;
+
+interface UploadResponse {
+    fileUploadResult: FileUploadResults;
+    file?: File;
+}
+export default async function uploader(
+    worker: any,
+    reader: FileReader,
+    existingFilesInCollection: File[],
+    rawFile: globalThis.File,
+    collection: Collection,
+): Promise<UploadResponse> {
+    UIService.setFileProgress(rawFile.name, 0);
+    let file: FileInMemory = null;
+    let encryptedFile: EncryptedFile = null;
+    try {
+        file = await UploadService.readFile(reader, rawFile, collection);
+
+        if (fileAlreadyInCollection(existingFilesInCollection, file)) {
+            // set progress to -2 indicating that file upload was skipped
+            UIService.setFileProgress(rawFile.name, FileUploadResults.SKIPPED);
+            await sleep(TwoSecondInMillSeconds);
```

Comment:
What's the reason for adding the 2 second sleep here? Can we add a comment mentioning the same?

Reply:
so that file remain in the progressbar screen for 2 second with the message - "duplicate file skipping..."